geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -215,49 +213,39 @@ object AccumuloDataStoreFactory {
 
   // Kerberos is only available in Accumulo >= 1.7.
   // Note: doesn't confirm whether correctly configured for Kerberos e.g. core-site.xml on CLASSPATH
-  def isKerberosAvailable: Boolean = {
+  def isKerberosAvailable: Boolean =
     AccumuloVersion.accumuloVersion != AccumuloVersion.V15 && AccumuloVersion.accumuloVersion != AccumuloVersion.V16
-  }
 
   def canProcess(params: JMap[String,Serializable]): Boolean = {
-
-    // Determine which parameters are set. Those which are required according to the parameter definition will throw an
-    // exception on lookup, so need the extra containsKey check.
-    val hasConnection = connParam.lookup[String](params) != null
-    val hasInstanceId = params.containsKey(instanceIdParam.key) && instanceIdParam.lookup[String](params) != null
-    val hasZookeepers = params.containsKey(zookeepersParam.key) && zookeepersParam.lookup[String](params) != null
-    val hasUser = params.containsKey(userParam.key) && userParam.lookup[String](params) != null
-    val hasPassword = params.containsKey(passwordParam.key) && passwordParam.lookup[String](params) != null
-    val hasKeytabPath = params.containsKey(keytabPathParam.key) && keytabPathParam.lookup[String](params) != null
-
-    val passwordAuth = hasInstanceId && hasZookeepers && hasUser && hasPassword && !hasKeytabPath
-    val kerberosAuth = hasInstanceId && hasZookeepers && hasUser && !hasPassword && hasKeytabPath && isKerberosAvailable
-
-    hasConnection || passwordAuth || kerberosAuth
+    val hasConnector = ConnectorParam.lookupOpt(params).isDefined
+    val hasConnection = InstanceIdParam.exists(params) && ZookeepersParam.exists(params) && UserParam.exists(params)
+    val hasPassword = PasswordParam.exists(params) && !KeytabPathParam.exists(params)
+    val hasKeytab = !PasswordParam.exists(params) && KeytabPathParam.exists(params) && isKerberosAvailable
+    hasConnector || (hasConnection && (hasPassword || hasKeytab))
```

Comment:
Idly, I wonder if we should make the val here lazy.:)

Reply:
i considered it, but @tkunicki says that lazy vals should be avoided due to memory flush, and since the calculations here aren't too complicated i left them broken out for readability

Reply:
How about changing them to def's?

Reply:
changed to defs